### PR TITLE
dasLLAMA: an untied Q8 load keeps its Q8_0 token table as quants (IMAGE_VERSION 33)

### PR DESCRIPTION
**Pull request by borisbat**
**Behavior change: `IMAGE_VERSION` moves to 33 - every prepared `.dlim` re-mints once on its next load.**

**Why.** An untied Q8 model kept its token-embedding table as f32 in `fblob`, vocab x dim floats: 37 MB on stories15M, 1.24 GB on the 152k-vocab Qwen3-30B-A3B. The flag that dropped the table (`cls_q8`) also meant "the tied classifier matmuls the Q8 quants", so a model with its own `output.weight` never took it.

**What changes.**
- `Model.emb_q8` names the embedding half alone: a Q8 load of a Q8_0 `token_embd` keeps the linear q8 copy at `emb_q8_off` and no f32 table, tied or untied; `cls_q8` keeps its classifier meaning (`cls_q8` = tied and `emb_q8`).
- The layout carves the copy in the untied branch, the loader skips the f32 read and transcodes `token_embd` beside `output.weight`.
- `embed_row`, the trim rail's `embq` build and the Metal blob eligibility gate read `emb_q8`; the image meta serializes it.
- The image rail's planar-to-blob rebake works again. Since the review-round fix batch of 2026-08-29 a metal-mode load that found a planar image mapped it and asked `metal_blob_eligible`, which refuses every mapped model, so the blob flavor was never rebaked and a GPU box served from the CPU planes for the rest of the process. The layout question is now `metal_blob_form_ok` (quant, packing flags, alignment); `metal_blob_eligible` keeps the in-place-transform refusal on top of it. The `metal` image arm was red on master for this; it caught the untied cell first.

**Observable behavior.**
- metal-mode load with a planar image already on disk -> the blob flavor is rebaked from the gguf and the GPU serves (was: planar mapped, CPU served, silently).
- untied Q8 load -> no f32 embedding table: stories15M's image 51 MB -> 26.6 MB; TinyLlama-1.1B drops 262 MB, Phi-3.5-mini 394 MB, Qwen3-30B-A3B 1.24 GB (sized, not measured). Tied models (Qwen3-4B, Llama-3.2, SmolLM2) never carried the table and are unchanged.
- any existing `.dlim` -> re-mints once on its next load (version 32 -> 33).
- Vulkan device embed on an untied q8 model -> the f32-upload arm declines, the CPU embed loop serves.

**Where to look.** `layout_offsets` and the untied `load_big` branch in `dasllama_load.das`, `embed_row` in `dasllama_common.das`; the rebake decision in `load_model_cached` (`dasllama_image.das`) and `metal_blob_form_ok` in `dasllama_layout.das`, with the rule in `ARCHITECTURE_IMAGE.md` 2.1n.

<details>
<summary>Validation, claims, ledger</summary>

### Validation
- `test_parity_untied_emb_q8_rows` (new): stories15M q8 rows bit-identical to the fp32 decode, no table kept, `embed_row` agrees.
- image suite arm `untied` (new): stories15M mint, map, token-for-token, image smaller than gguf plus an f32 table.
- image suite arm `metal-untied` (new): TinyLlama-1.1B on the Metal blob flavor - the layout gate admits the untied emb_q8 copy, cold mint + warm map keep the q8 table with no f32 copy, GPU serve, teacher-forced logits vs the CPU control (GPU greedy picks equal the CPU continuation on the counting prompt).
- image suite arm `metal` (existing): red before the rebake fix, green after - its CPU control mints the planar image first, which is the shape the rebake serves.
- The full preflight's fast tier went red once on lint: master's fresh jobque tripwire logs from a hot-path callee, which the hot-path lint walked into every kernel. Fixed here by marking that first-dispatch check a cold path; lint rerun green on all three rails. The lane gates (docs, tests-cpp, interp, JIT, AOT) did not rerun locally and are CI's; the dasLLAMA module gates (`model-free`, `stocked`) ran locally.
- Not run: Vulkan (no device here).

### Claims - stated, not tested
- Vulkan's device-side embed gather has no untied-q8 arm; the f32-upload arm now declines on `emb_q8` and the CPU embed loop serves. A break would show as a prefill on an untied q8 model under `DASLLAMA_VK_GPU_EMBED` reading an f32 table that is no longer there.

### Not done
- A Vulkan gather over the linear q8 copy for untied models.

</details>

### CODE REVIEW

File: modules/dasLLAMA/tests/test_model_image.das [hunk 1]
```diff
@@ -490,6 +490,26 @@ def test_model_image_roundtrip(t : T?) {
...
+        let path = path_join(models_dir(), "stories15M-Q8_0.gguf")
+        if (!model_available(t, path)) {
+            return
+        }
+        text_image_roundtrip(t, path, 24)
+        var m <- load_model(path, QuantMode.q8)
+        t |> success(m.emb_q8 && !m.cls_q8 && m.tok_emb_off < 0l, "the mapped image serves the untied q8 table with no fp32 copy")
+        let img_bytes = int64(stat(image_path_for(path)).size)
+        let table_f32 = m.config.vocab_size * m.config.dim * 4l
+        t |> success(img_bytes < table_f32 + int64(stat(path).size), "the image ({img_bytes >> 20l} MB) is smaller than the gguf plus an f32 table: no f32 table rode along")
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/tests/test_model_image.das]: Accepted. The bar was indeed inert: an image that still carried the table measured 51 MB against a gguf-plus-table bound of 63 MB, so it would have passed. The assertion is now `img_bytes < table_f32` - the whole image must be smaller than the f32 table alone (27 MB against 37 MB on stories15M); an image carrying the table cannot be. Re-ran the `untied` image arm on the fixed tip (82f583b5c): green.

File: modules/dasLLAMA/dasllama/dasllama_layout.das [hunk 2]
```diff
@@ -97,7 +97,7 @@ def private metal_blob_layout_ok(t : Model) : bool {
     }
     let cls_fmt = c.shared_weights ? t.emb_fmt : t.wcls_fmt
     return (metal_blob_off_ok(t.wcls_off, cls_fmt) &&
-        (!t.cls_q8 || metal_blob_off_ok(t.emb_q8_off, KqFmt.q8)))
+        (!t.emb_q8 || metal_blob_off_ok(t.emb_q8_off, KqFmt.q8)))
 }
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/dasllama/dasllama_layout.das:101]: Added in d303946de: image arm `metal-untied` on TinyLlama-1.1B (the untied Q8 the Metal drivers serve - stories15M is untied too, but its 48-wide heads fail the Metal shape gate). The cell pins the CPU control, cold-mints the blob flavor, asserts `emb_q8 && !cls_q8 && tok_emb_off < 0` with `fblob` smaller than the table on both the cold mint and the warm map, generates through the GPU, and holds teacher-forced logits against the planar CPU control.

Running it exposed a master defect the `metal` arm had been red on since the 2026-08-29 fix batch: a metal load that finds a planar image on disk asked `metal_blob_eligible`, which refuses every mapped model, so the blob flavor was never rebaked and the process served from the CPU planes. Fixed in 06b67fa2a - the layout question is now `metal_blob_form_ok`, `metal_blob_eligible` keeps the in-place refusal on top of it; rule in `ARCHITECTURE_IMAGE.md` 2.1n. Both arms green locally.
